### ISSUE TITLE
Model field (enum via $casts) has type mixed

### ISSUE THREAD

**Issue by levu42** [nunomaduro/larastan#1505]
- Larastan Version: 2.3.4
- `--level` used: 8

### Description

I have a model where one column is represented as a PHP Enum (string backed enum with currently only one possible option), declared via $casts, but PHPStan reports the type incorrectly as `mixed`.

### Laravel code where the issue was found

```php

enum MyModelType: string
{
    case Foo = 'foo';
}

class MyModel {
    protected $casts = [
        'type' => MyModelType::class,
    ];
}

class Bar {
    function baz(MyModel $m): int
    {
        return match($m->type) {
            MyModelType::Foo => 42,
        };
    }
}
```

I get now the error message `Match expression does not handle remaining value: mixed` and when I use `\PHPStan\dumpType($m->type)` I get `Dumped type: mixed` instead of the expected `MyModelType`.

It seems like this only starts happening since I upgraded from 2.3.0 -> 2.3.4 but I'm not 100% sure about that.

**Comment by canvural** [comment]
This might be fixed with #1494

**Comment by levu42** [comment]
@canvural Awesome!
